sphinx/writers/latex.py

```diff
@@ -2163,26 +2147,36 @@ def depart_title_reference(self, node):
         # type: (nodes.Node) -> None
         self.body.append('}')
 
+    def visit_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        longest_label = max((subnode[0].astext() for subnode in node), key=len)
+        self.body.append(u'\n\\begin{sphinxthebibliography}{%s}\n' % longest_label)
+
+    def depart_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        self.body.append(u'\\end{sphinxthebibliography}\n')
+
     def visit_citation(self, node):
         # type: (nodes.Node) -> None
-        # TODO maybe use cite bibitems
-        # bibitem: [citelabel, citetext, docname, citeid]
-        self.bibitems.append(['', '', '', ''])
-        self.context.append(len(self.body))
+        label = node[0].astext()
+        self.body.append(u'\\bibitem[%s]{%s:%s}{' %
+                         (self.encode(label), node['docname'], node['ids'][0]))
 
     def depart_citation(self, node):
         # type: (nodes.Node) -> None
-        size = self.context.pop()
-        text = ''.join(self.body[size:])
-        del self.body[size:]
-        self.bibitems[-1][1] = text
+        self.body.append('}\n')
 
     def visit_citation_reference(self, node):
         # type: (nodes.Node) -> None
-        # This is currently never encountered, since citation_reference nodes
-        # are already replaced by pending_xref nodes in the environment.
-        self.body.append('\\cite{%s}' % self.idescape(node.astext()))
-        raise nodes.SkipNode
+        if self.in_title:
+            pass
```

Comment:
Note: On my local, `\cite` does not work well in section title. So I disable references inside title node. It means they are also disabled title of tables, title of admonitions and so on.

Reply:
ok. I'll investigate later how to handle this.